extensions/websockets-next/deployment/src/main/java/io/quarkus/websockets/next/deployment/WebSocketProcessor.java

```diff
@@ -386,6 +388,22 @@ void collectEndpoints(BeanArchiveIndexBuildItem beanArchiveIndex,
                         "The endpoint must declare at least one method annotated with @OnTextMessage, @OnBinaryMessage, @OnPingMessage, @OnPongMessage or @OnOpen: "
                                 + beanClass);
             }
+            if (onTextMessage != null && onTextMessage.messageType() == MessageType.TEXT) {
```

Comment:
```suggestion
            if (onTextMessage != null) {
```
`&& onTextMessage.messageType() == MessageType.TEXT` is actually redundant...